src/Agent.Worker/Handlers/StepHost.cs

```diff
@@ -97,12 +98,15 @@ public string ResolvePathForStepHost(string path)
             // remove double quotes around the path
             path = path.Trim('\"');
 
+            // if running on OS X, just translate the path
+            if (PlatformUtil.RunningOnMacOS)
+            {
+                return Container.TranslateToContainerPath(path);
+            }
```

Comment:
This file and externals.sh are the only ones with functional changes, right?

One nit - as a reviewer I think it would be helpful to either pull the refactoring changes into a different PR or just call out where the functional changes are happening. That makes it easier to tell what is going on.

With that said, I do love the refactoring changes ❤️

Reply:
I think there were other files with logic changes that I ended up backing out while triaging, but I agree with you and usually try to not mix refactoring with logic changes.